### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by tanjia123456** [KDDGRAND/grand#8]
Hello, thank you for your works.
Now, I want to apply it to my dataset, I split the def train function into two parts: training and testing, but now I report an error: segmentation fault (core damped)
Could you help me to see what's wrong?
`
def train_before(features,labels):
    X = features
    model.train()
    optimizer.zero_grad()
    X_list = []
    K = args.sample
    for k in range(K):
        X_list.append(rand_prop(X, training=True))
    output_list = []
    for k in range(K):
        output_list.append(torch.log_softmax(model(X_list[k]), dim=-1))
    loss_train = 0.
    for k in range(K):
        loss_train += F.nll_loss(output_list[k], labels)

    loss_train = loss_train / K
    loss_consis = consis_loss(output_list)

    loss_train = loss_train + loss_consis
    acc_train = accuracy(output_list[0], labels)
    loss_train.backward()
    optimizer.step()

    return loss_train,acc_train

def val_before(features,labels):
    X = features
    model.eval()
    X = rand_prop(X, training=False)
    output = model(X)
    output = torch.log_softmax(output, dim=-1)

    loss_val = F.nll_loss(output, labels)
    acc_val = accuracy(output, labels)

    return loss_val,acc_val

`

thank you very much
